### PR TITLE
Use llvm project

### PR DESCRIPTION
**Pull request by lzydmxy**
### Which issues of this PR fixes:

This PR is part of #291  
Replace the ClickHouse‘s `libcxx` and `libcxxabi`  with llvm-project.
Can't Replace ClickHouse‘s `libunwind` now, because jemalloc still need `unw_backtrace` , and our stackTrace also use `unw_backtrace`
### Change log:

Add llvm-project 17 to instead of ClickHouse‘s `libcxx` and `libcxxabi`

### CODE REVIEW

File: src/Service/FourLetterCommand.cpp [hunk 1]
```diff
@@ -45,9 +45,10 @@ String IFourLetterCommand::toName(int32_t code)
 
 int32_t IFourLetterCommand::toCode(const String & name)
 {
-    int32_t res = *reinterpret_cast<const int32_t *>(name.data());
+    int32_t res;
+    std::memcpy(&res, name.data(), sizeof(res));
     /// keep consistent with Coordination::read method by changing big endian to little endian.
-    return __builtin_bswap32(res);
```
Comment (JackyWoo) [inline src/Service/FourLetterCommand.cpp:50]: why not use` __builtin_bswap32`

Comment (lzydmxy) [reply src/Service/FourLetterCommand.cpp:50]: std::byteswap is standardized in C++23, it better than use __builtin_bswap32 which is a GCC/Clang built-in function.

**Comment by lzydmxy** [comment]
I will open another pr to fix failed test #363

File: src/Service/FourLetterCommand.cpp (hunk 1 above)
Comment (JackyWoo) [reply src/Service/FourLetterCommand.cpp:50]: Please find and replace `__builtin_bswap32 ` in global

File: base/common/logger_useful.h [hunk 2]
```diff
@@ -13,6 +13,11 @@ namespace
     template <typename T, typename... Ts> constexpr auto firstArg(T && x, Ts &&...) { return std::forward<T>(x); }
 }
 
+template<typename... Args>
```
Comment (JackyWoo) [inline base/common/logger_useful.h:16]: It seems `logFormat` is meaningless.

File: src/Service/FourLetterCommand.cpp (hunk 1 above)
Comment (lzydmxy) [reply src/Service/FourLetterCommand.cpp:50]: okay

File: src/Service/KeeperStore.cpp [hunk 3]
```diff
@@ -1018,7 +1019,7 @@ struct StoreRequestMultiTxn final : public StoreRequest
             }
             else
                 throw RK::Exception(
-                    ErrorCodes::BAD_ARGUMENTS, "Illegal command as part of multi ZooKeeper request {}", sub_zk_request->getOpNum());
+                    ErrorCodes::BAD_ARGUMENTS, "Illegal command as part of multi ZooKeeper request {}", fmt::underlying(sub_zk_request->getOpNum()));
```
Comment (JackyWoo) [inline src/Service/KeeperStore.cpp:1022]: Please toString(sub_zk_request->getOpNum())

File: src/Service/KeeperStore.cpp [hunk 4]
```diff
@@ -1153,7 +1154,7 @@ class StoreRequestFactory final : private boost::noncopyable
     void registerRequest(Coordination::OpNum op_num, Creator creator)
     {
         if (!op_num_to_request.try_emplace(op_num, creator).second)
-            throw RK::Exception(ErrorCodes::LOGICAL_ERROR, "Request with op num {} already registered", op_num);
+            throw RK::Exception(ErrorCodes::LOGICAL_ERROR, "Request with op num {} already registered", fmt::underlying(op_num));
```
Comment (JackyWoo) [inline src/Service/KeeperStore.cpp:1157]: Same with the previous one

File: src/Service/NuRaftLogSnapshot.cpp [hunk 5]
```diff
@@ -897,7 +897,7 @@ size_t KeeperSnapshotManager::removeSnapshots()
 {
     Int64 remove_count = static_cast<Int64>(snapshots.size()) - static_cast<Int64>(keep_max_snapshot_count);
 
-    LOG_INFO(log, "There are {} snapshots, we will try to move {remove_count} of them", snapshots.size(), remove_count);
```
Comment (JackyWoo) [inline src/Service/NuRaftLogSnapshot.cpp:900]: This is truly a trivial mistake

File: src/Service/NuRaftStateMachine.cpp [hunk 6]
```diff
@@ -408,7 +408,7 @@ void NuRaftStateMachine::replayLogs(ptr<log_store> log_store_, uint64_t from, ui
                 {
                     if (entry_with_version.entry->get_val_type() != nuraft::log_val_type::app_log)
                     {
-                        LOG_DEBUG(thread_log, "Found non app nuraft log(type {}), ignore it", entry_with_version.entry->get_val_type());
+                        LOG_DEBUG(thread_log, "Found non app nuraft log(type {}), ignore it", fmt::underlying(entry_with_version.entry->get_val_type()));
```
Comment (JackyWoo) [inline src/Service/NuRaftStateMachine.cpp:411]: It is better to add a `toString` method

**Comment by JackyWoo** [comment]
@lzydmxy now the CI has a little problem which I fixed in #362

**Comment by JackyWoo** [comment]
> will open another pr to fix failed test #363

ok

**Comment by JackyWoo** [comment]
Do not forget to update the min clang version in readme and how-to-build docs.

**Review by JackyWoo — APPROVED** [review]

**Pull request merged**